packages/features/ee/workflows/components/WorkflowStepContainer.tsx

```diff
@@ -133,37 +130,25 @@ export default function WorkflowStepContainer(props: WorkflowStepProps) {
   const { data: actionOptions } = trpc.viewer.workflows.getWorkflowActionOptions.useQuery();
   const triggerOptions = getWorkflowTriggerOptions(t);
   const templateOptions = getWorkflowTemplateOptions(t, step?.action, hasActiveTeamPlan);
+  if (step && !form.getValues(`steps.${step.stepNumber - 1}.reminderBody`)) {
+    const action = form.getValues(`steps.${step.stepNumber - 1}.action`);
+    const template = getTemplateBodyForAction({
+      action,
+      locale: i18n.language,
+      template: step.template ?? WorkflowTemplates.REMINDER,
+      timeFormat,
+    });
+    form.setValue(`steps.${step.stepNumber - 1}.reminderBody`, template);
+  }
 
-  if (step && form.getValues(`steps.${step.stepNumber - 1}.template`) === WorkflowTemplates.REMINDER) {
```

Comment:
This type of logic was repeated through out the component. I decided to abstract this to `getTemplateBodyForAction` so it can be be used in other parts of the codebase